src/server/wasm/wasm_registry.cc

```diff
@@ -37,17 +41,28 @@ WasmRegistry::WasmRegistry()
   store_.context().set_wasi(std::move(wasi)).unwrap();
 
   linker_.define_wasi().unwrap();
+
+  InstantiateAndLinkModules();
 }
 
 WasmRegistry::~WasmRegistry() {
 }
 
+void WasmRegistry::InstantiateAndLinkModules() {
+  auto wasm_modules = absl::GetFlag(FLAGS_wasmpaths);
+  std::vector<std::string_view> modules = absl::StrSplit(wasm_modules, ",");
+  for (auto mod_path : modules) {
+    Add(mod_path);
+  }
+}
+
 std::string WasmRegistry::Add(std::string_view path) {
   // 1. Read the wasm file in path
   auto is_file_read = io::ReadFileToString(path);
   if (!is_file_read) {
-    return absl::StrCat("File error for path: ", path, " with error ",
-                        is_file_read.error().message());
+    LOG(ERROR) << "File error for path: " << path << " with error "
+               << is_file_read.error().message();
+    exit(1);
```

Comment:
For now we should exit, I don't want any of the Engine's instantiated if the modules failed to load. Later we will relax this as we add support for loading modules at runtime (once we resolve the issue with the SEGFAULT in the scheduler's switch fiber`